presto-orc/src/main/java/com/facebook/presto/orc/reader/StructStreamReader.java

```diff
@@ -196,6 +184,28 @@ public String toString()
                 .toString();
     }
 
+    private Block[] getBlocksForType(Type type, int positionCount) throws IOException
+    {
+        RowType rowType = (RowType) type;
+
+        Block[] blocks = new Block[rowType.getFields().size()];
+
+        for (int i = 0; i < rowType.getFields().size(); i++) {
+            Optional<String> fieldName = rowType.getFields().get(i).getName();
```

Comment:
IIRC there was some issue with case sensitivity when we added this feature to the Parquet reader.  I think the issue was that Presto lower cases the field names in the type, so we may have to lower case when building the `structFields` map in the constructor. cc @nezihyigitbasi @zhenxiao for context on Parquet